src/main/java/org/opentripplanner/routing/algorithm/ItineraryRouters.java

```diff
@@ -0,0 +1,363 @@
+
+    private Instant filterOnLatestDepartureTime = null;
+    private int searchWindowUsedInSeconds = NOT_SET;
+
+    public ItineraryRouters(
+            RoutingRequest request,
+            Router router,
+            DebugTimingAggregator debugTimingAggregator,
+            RaptorService<TripSchedule> raptorService
+    ) {
+        this.request = request;
+        this.router = router;
+        this.debugTimingAggregator = debugTimingAggregator;
+        this.raptorService = raptorService;
+
+        this.itineraries = Collections.synchronizedList(new ArrayList<>());
+        this.routingErrors = Collections.synchronizedList(new ArrayList<>());
+    }
+
+    public List<Itinerary> getItineraries() {
+        return itineraries;
+    }
+
+    public List<RoutingError> getRoutingErrors() {
+        return routingErrors;
+    }
+
+    public Instant getFilterOnLatestDepartureTime() {
+        return filterOnLatestDepartureTime;
+    }
+
+    public int getSearchWindowUsedInSeconds() {
+        return searchWindowUsedInSeconds;
+    }
+
+    public void collectItineraries() {
+        if (OTPFeature.ParallelRouting.isOn()) {
+            CompletableFuture.allOf(
+                    CompletableFuture.runAsync(this::routeDirectStreet),
+                    CompletableFuture.runAsync(this::routeDirectFlex),
+                    CompletableFuture.runAsync(this::routeTransit)
+            ).join();
+        } else {
+            // Direct street routing
+            routeDirectStreet();
+
+            // Direct flex routing
+            routeDirectFlex();
+
+            // Transit routing
+            routeTransit();
+        }
+    }
+
+    public void routeDirectStreet() {
+        debugTimingAggregator.startedDirectStreetRouter();
+        try {
+            itineraries.addAll(DirectStreetRouter.route(router, request));
+        } catch (RoutingValidationException e) {
+            routingErrors.addAll(e.getRoutingErrors());
+        } finally {
+            debugTimingAggregator.finishedDirectStreetRouter();
+        }
+    }
+
+    public void routeDirectFlex() {
+        if (!OTPFeature.FlexRouting.isOn()) {
+            return;
+        }
+
+        debugTimingAggregator.startedDirectFlexRouter();
+        try {
+            itineraries.addAll(DirectFlexRouter.route(router, request));
+        }
+        catch (RoutingValidationException e) {
+            routingErrors.addAll(e.getRoutingErrors());
+        } finally {
+            debugTimingAggregator.finishedDirectFlexRouter();
+        }
+    }
+
+    public void routeTransit() {
+        debugTimingAggregator.startedTransitRouting();
+        try {
+            itineraries.addAll(doRouteTransit());
+        }
+        catch (RoutingValidationException e) {
+            routingErrors.addAll(e.getRoutingErrors());
+        } finally {
+            debugTimingAggregator.finishedTransitRouter();
+        }
+    }
+
+    private Collection<Itinerary> doRouteTransit() {
```

Comment:
Should this be pushed into a `TransitRouter`, similar to how `DirectStreetRouter` and `DirectFlexRouter` are?

Reply:
:100: , I've created an explicit `TransitRouter`